### PR TITLE
Implemented: code to enable preorder app on the product cards having add to cart button(#1vd3hp4)

### PR DESCRIPTION
**Pull request by ymaheshwari1**
- Requirements:
  - Need to add two input fields with type hidden, one having product id (virtual id) and the other having tags information with name productId and tags respectively on the product card component.
 
- Defined a separate method to handle add to cart from the product card, as having an overriding issue, also remove the usage of global variable as the variables are getting overridden and having unexpected behavior.

Created branch from #22

### CODE REVIEW

File: scripts/script-tag.js [hunk 1]
```diff
@@ -147,18 +151,60 @@
...
                 // will handle the click event on the pre order button
                 preorderButton.on("click", addToCart);
             }
+        } else {
+            // this part executes on all the page other than product page
+            // finding an input field with name tags and then iterating over the same
+            jQueryPreOrder("input[name='tags']").map(async function (index, element) {
+                // finding a button with type submit as the button will be on the same level as the input field so using siblings
+                const preorderButton = jQueryPreOrder(element).siblings("input[type='submit']");
+                preorderButton.val(theme.strings.addToCart)
```
Comment (adityasharma7) [inline scripts/script-tag.js:168]: Why we are using theme object here?

Comment (ymaheshwari1) [reply scripts/script-tag.js:168]: Sir, to get the text of add to cart button, and whenever coming to the collections or any other page first we make it the default text for the button.

File: scripts/script-tag.js [hunk 2]
```diff
@@ -147,16 +148,92 @@
...
                 // will handle the click event on the pre order button
                 preorderButton.on("click", addToCart);
             }
+        } else {
+            // this part executes on all the page other than product page
+            // finding an input field with name tags and then iterating over the same
+            jQueryPreOrder("input[name='tags']").map(async function (index, element) {
+
+                // checking for Pre-Orde or Back-Order tag
+                if (jQueryPreOrder(element).val().includes('Pre-Order') || jQueryPreOrder(element).val().includes('Back-Order')) {
```
Comment (adityasharma7) [inline scripts/script-tag.js:165]: I think we should assign the value to a variable and then use further

File: scripts/script-tag.js [hunk 3]
```diff
@@ -147,16 +148,92 @@
...
+                    // getting the current display variant id and current virtual product id
+                    const variantId = jQueryPreOrder(element).siblings("input[name='id']").val();
```
Comment (adityasharma7) [inline scripts/script-tag.js:167]: I think we should assign jQueryPreOrder(element) to a variable and then use further

File: scripts/script-tag.js [hunk 4]
```diff
@@ -147,16 +148,92 @@
...
+                    const virtualId = jQueryPreOrder(element).siblings("input[name='productId']").val();
+
+                    await isItemAvailableForOrder(virtualId, variantId).then(async (product) => {
+                        // checking what type of tag product contains (Pre-Order / Back-order) and on the basis of that will check for metafield
+                        const productType = product.tags.includes('Pre-Order') ? 'Pre-Order' : product.tags.includes('Back-Order') ? 'Back-Order' : ''
+
+                        // checking if continue selling is enabled for the variant or not
+                        const checkItemAvailablity = product.variants.find((variant) => variant.id == variantId).inventory_policy === 'continue'
```
Comment (adityasharma7) [inline scripts/script-tag.js:175]: Improve variable name

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: scripts/script-tag.js (hunk 3 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js:167]: Sir, can we name it `const tagInput = jQueryPreOrder(element)`? as the element is input with name tag.

Comment (adityasharma7) [reply scripts/script-tag.js:167]: It should be close to what it stores variantIdInput

File: scripts/script-tag.js (hunk 1 above)
Comment (adityasharma7) [reply scripts/script-tag.js]: Is it a generic solution?

File: scripts/script-tag.js (hunk 3 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js:167]: Sir, it stores the input field having tags, `variantTagInput`, WDYT?

File: scripts/script-tag.js (hunk 1 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: I have remove it after updating the code to simplify the logic.

File: scripts/script-tag.js (hunk 3 above)
Comment (adityasharma7) [reply scripts/script-tag.js:167]: LGTM

Comment (ymaheshwari1) [reply scripts/script-tag.js]: Done Sir

File: scripts/script-tag.js (hunk 4 above)
Comment (ymaheshwari1) [reply scripts/script-tag.js]: Done Sir

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**